src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/TypeOperations.cs

```diff
@@ -2460,7 +2460,7 @@ protected internal override Expression VisitLambda<T>(Expression<T> node)
                 {
                     return node;
                 }
-                return node.Update(b, node.Parameters);
+                return Expression.Lambda<T>(b, node.Name, node.TailCall, node.Parameters);
```

Comment:
Same as `ExpressionQuoter` used by the compiler; we know the `Body` changed, so `Update` will call the factory anyway. I'm wondering whether there was any particular reason for this change here when the code was cloned? Any restrictions with .NET Native / AOT we're aware of? I can also converge both copies of the code by calling `Update` in both places; it'd be no big deal.